Vagrantfile

```diff
@@ -195,8 +195,13 @@ Vagrant.configure("2") do |config|
       inline: BOOTSTRAP
   end
 
-  (1..9).each do |i|
-    node_name = "node#{i}"
+  INSTALL_PYTHON = <<-SCRIPT
```

Comment:
Let's put this nearer to the relevant loop (i.e. just before the `ubuntu` loop).

___

Alternatively, we could have nested loops roughly looking like this (untested):

```rb
os_data = {
  "ubuntu" => {
    "name" => "ubuntu/bionic",
    "version" => "20190514.0.0"
  },
  "centos" => {
    "name" => "centos/7",
    "version" => "1811.02"
  }
}

os_data.each do |os, os_data|
  (1..5).each do |i|
    if os == "ubuntu"
      # Do your script
    end
    config.vm.box = os_data["name"]
    config.vm.version = os_data["version"]

    # etc etc
  end
end
```
NBD if you want to keep your version, it's fine as is.

Reply:
+1

I prefer loop over copy-pasta, so that if one day we decide to add a settings or change an existing one (like `node.vbguest.auto_update`) we will have a single place to update instead of many.

Reply:
loop now used for node deployment.